src/source/Ice/TurnConnection.c

```diff
@@ -160,6 +160,10 @@ STATUS turnConnectionIncomingDataHandler(UINT64 customData, PSocketConnection pS
         CHK_STATUS(turnConnectionHandleChannelDataTcpMode(pTurnConnection, pBuffer, bufferLen));
     }
 
+    // Unlock before checking if need to deliver to data channel callback
+    MUTEX_UNLOCK(pTurnConnection->lock);
```

Comment:
I guess this is typo, mutex is released in cleanup, a couple of lines below.

Reply:
Yes, it does look weird to do it again. Let me change this. Thanks for spotting it.